### PR TITLE
chore: upgrade StopTimeTimepointWithoutTimesNotice to ERROR

### LINKED ISSUE #1017: Split PR 970 into one PR per rule.
Rules left to be upgraded to error:
-  LocationWithUnexpectedStopTimeNotice
-  StopTimeTimepointWithoutTimesNotice (bans 230 feeds at Google)
-  EmptyColumnNameNotice
-  StopWithoutZoneId (we need stats)
-  PathwaytUnreachableLocationNotice
- TranslationForeignKeyViolationNotice
- MissingLevelIdNotice (we need stats)

### PR DESCRIPTION
**Pull request by lionel-nj**
closes #1017 
[More context](https://github.com/MobilityData/gtfs-validator/pull/1022)
**Summary:**

This PR provides support to upgrade `StopTimeTimepointWithoutTimeNotice` to ERROR.

**Expected behavior:** 

- `StopTimeTimepointWithoutTimeNotice`: 

> |  Field Name | Type | Presence | Description |
> |  ------ | ------ | ------ | ------ |
> |  `trip_id` | ID referencing `trips.trip_id` | **Required** | Identifies a trip.  |
> |  `arrival_time` | Time | **Conditionally Required** | Arrival time at the stop (defined by `stop_times.stop_id`) for a specific trip (defined by `stop_times.trip_id`). <br><br>If there are not separate times for arrival and departure at a stop, `arrival_time` and `departure_time` should be the same. <br><br>For times occurring after midnight on the service day, enter the time as a value greater than 24:00:00 in HH:MM:SS local time for the day on which the trip schedule begins.<br><br> If exact arrival and departure times (`timepoint=1` or empty) are not available, estimated or interpolated arrival and departure times (`timepoint=0`) should be provided.<br><br>Conditionally Required:<br>- **Required** for the first and last stop in a trip (defined by `stop_times.stop_sequence`). <br>- **Required** for `timepoint=1`.<br>- Optional otherwise.|
> |  `departure_time` | Time | **Conditionally Required** | Departure time from the stop (defined by `stop_times.stop_id`) for a specific trip (defined by `stop_times.trip_id`).<br><br>If there are not separate times for arrival and departure at a stop, `arrival_time` and `departure_time` should be the same. <br><br>For times occurring after midnight on the service day, enter the time as a value greater than 24:00:00 in HH:MM:SS local time for the day on which the trip schedule begins.<br><br> If exact arrival and departure times (`timepoint=1` or empty) are not available, estimated or interpolated arrival and departure times (`timepoint=0`) should be or
> 

_https://github.com/google/transit/blob/master/gtfs/spec/en/reference.md#stop_timestxt_

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `gradle test` to make sure you didn't break anything
- [x] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [x] Linked all relevant issues
- ~[ ] Include screenshot(s) showing how this pull request works and fixes the issue(s)~

### CODE REVIEW

**Comment by lionel-nj** [comment]
This notice is emitted by 44 datasets from the [MobilityDatabase](http://mobilitydatabase.org/wiki/Main_Page).

**Review by isabelle-dr — CHANGES_REQUESTED** [review]
Could you update docs/MIGRATION_V2_V3.md as well?
Besides, LGTM, thanks!

File: main/src/main/java/org/mobilitydata/gtfsvalidator/validator/TimepointTimeValidator.java [hunk 1]
```diff
@@ -93,7 +92,7 @@ private boolean isTimepoint(GtfsStopTime stopTime) {
     private final String specifiedField;
 
     StopTimeTimepointWithoutTimesNotice(GtfsStopTime stopTime, String specifiedField) {
-      super(SeverityLevel.WARNING);
```
Comment (asvechnikov2) [inline main/src/main/java/org/mobilitydata/gtfsvalidator/validator/TimepointTimeValidator.java:96]: Just writing my thoughts, no need to change anything. The fact that changing a severity level doesn't require changing any tests feels a bit off, not sure if it's a good thing or not.

**Review by asvechnikov2 — APPROVED** [review]
A couple of comments, otherwise, LGTM!

File: main/src/main/java/org/mobilitydata/gtfsvalidator/validator/TimepointTimeValidator.java (hunk 1 above)
Comment (lionel-nj) [reply main/src/main/java/org/mobilitydata/gtfsvalidator/validator/TimepointTimeValidator.java:96]: I added that to the [list of points ](https://github.com/MobilityData/gtfs-validator/issues/1052)to discuss regarding how our unit tests are set up.

File: docs/MIGRATION_V2_V3.md [hunk 2]
```diff
@@ -82,7 +82,7 @@ New rules were added, others were replaced by several new rules, and others had
...
-| StopTimeTimepointWithoutTimesNotice                  | WARNING  | [StopTimeTimepointWithoutTimesNotice](https://github.com/MobilityData/gtfs-validator/blob/master/RULES.md#stoptimetimepointwithouttimesnotice)                                   | TBD      | This rule was improved to solve false positives                                                                                                                  |
+| StopTimeTimepointWithoutTimesNotice                  | ERROR    | [StopTimeTimepointWithoutTimesNotice](https://github.com/MobilityData/gtfs-validator/blob/master/RULES.md#stoptimetimepointwithouttimesnotice)                                   | TBD      | This rule was improved to solve false positives                                                                                                                  |
```
[the remark this answers is not in the document]
Comment (lionel-nj) [reply docs/MIGRATION_V2_V3.md]: Modified in https://github.com/MobilityData/gtfs-validator/pull/1068/commits/9dbe779934f43374f6761a1611f7cc73803c15a2.

**Comment by lionel-nj** [comment]
Thanks for reviewing @asvechnikov2!

**Review by maximearmstrong — APPROVED** [review]
LGTM!

**Pull request merged**